src/compiler/scala/tools/nsc/transform/Fields.scala

```diff
@@ -513,51 +513,60 @@ abstract class Fields extends InfoTransform with ast.TreeDSL with TypingTransfor
       * Desugar a local `lazy val x: Int = rhs` into
       * ```
       * val x$lzy = new scala.runtime.LazyInt()
-      * def x(): Int =
+      * def x$lzycompute(): Int =
       *   x$lzy.synchronized {
-      *     if (!x$lzy.initialized) {
+      *     if (x$lzy.initialized()) x$lzy.value()
+      *     else {
       *       x$lzy.initialized = true
       *       x$lzy.value = rhs
       *     }
-      *     x$lzy.value
       *  }
+      * def x(): Int = if (x$lzy.initialized()) x$lzy.value() else x$lzycompute()
       * ```
       */
     private def mkLazyLocalDef(lazyVal: Symbol, rhs: Tree): Tree = {
+      import CODE._
+      val owner = lazyVal.owner
+
       val lazyValType = lazyVal.tpe.resultType
-      val refClass = lazyHolders.getOrElse(lazyValType.typeSymbol, LazyRefClass)
-      val refTpe = if (refClass != LazyRefClass) refClass.tpe else appliedType(refClass.typeConstructor, List(lazyValType))
-
-      val flags = (lazyVal.flags & FieldFlags | ARTIFACT | MUTABLE) & ~(IMPLICIT | STABLE) // TODO: why include MUTABLE???
-      val name  = lazyVal.name.toTermName.append(nme.LAZY_LOCAL_SUFFIX_STRING)
-      val holderSym =
-        lazyVal.owner.newValue(name, lazyVal.pos, flags) setInfo refTpe
-
-      val accessor = mkAccessor(lazyVal) {
-        import CODE._
-        val initializedGetter = refTpe.member(nme.initialized)
-        val setInitialized    = Apply(Select(Ident(holderSym), initializedGetter.setterIn(refClass)), TRUE :: Nil)
-        val isUnit = refClass == LazyUnitClass
-        val valueGetter = if (isUnit) NoSymbol else refTpe.member(nme.value)
-        val valueSetter = if (isUnit) NoSymbol else valueGetter.setterIn(refClass)
-        val setValue    = if (isUnit) rhs      else Apply(Select(Ident(holderSym), valueSetter), rhs :: Nil)
-        val getValue    = if (isUnit) UNIT     else Apply(Select(Ident(holderSym), valueGetter), Nil)
-
-        // must read the value within the synchronized block since it's not volatile
-        // (there's no happens-before relation with the read of the volatile initialized field)
-        // TODO: double-checked locking (https://github.come/scala/scala-dev/issues/204)
-        gen.mkSynchronized(Ident(holderSym))(
-          If(Ident(holderSym) DOT initializedGetter, getValue, Block(List(setValue, setInitialized), getValue)))
+      val refClass    = lazyHolders.getOrElse(lazyValType.typeSymbol, LazyRefClass)
+      val refTpe      = if (refClass != LazyRefClass) refClass.tpe else appliedType(refClass.typeConstructor, List(lazyValType))
+
+      val lazyName  = lazyVal.name.toTermName
+      val pos       = lazyVal.pos.focus
+      val flags     = (lazyVal.flags & FieldFlags | ARTIFACT | MUTABLE) & ~(IMPLICIT | STABLE) // TODO: why include MUTABLE???
```

Comment:
the `MUTABLE` flag causes an ObjectRef to be created in lambdalift, and we get

```
    public int t1() {
        ObjectRef v$lzy = ObjectRef.create((Object)new LazyInt());
        return this.v$1(v$lzy);
    }

    private final int v$1(ObjectRef v$lzy$1) {
        return ((LazyInt)v$lzy$1.elem).initialized() ? ((LazyInt)v$lzy$1.elem).value() : this.v$lzycompute$1(v$lzy$1);
    }

    private final /* synthetic */ int v$lzycompute$1(ObjectRef v$lzy$1) {
      ...
    }
```

Removing the flag fixes that issue. Can we just do that, or will it cause other problems?